src/unit_tests/wazuh_db/test_wdb_global_parser.c

```diff
@@ -2295,6 +2295,150 @@ void test_wdb_parse_global_get_agents_by_connection_status_query_success(void **
     assert_int_equal(ret, OS_SUCCESS);
 }
 
+
+/* wdb_parse_global_get_backup */
+
+void test_wdb_parse_global_get_backup_failed(void **state) {
+    test_struct_t *data  = (test_struct_t *)*state;
+    int result = OS_INVALID;
+    char *query = NULL;
+
+    os_strdup("global backup get", query);
+    will_return(__wrap_wdb_open_global, data->wdb);
+    expect_string(__wrap__mdebug2, formatted_msg, "Global query: backup get");
+
+    will_return(__wrap_wdb_global_get_backups, NULL);
+
+    result = wdb_parse(query, data->output, 0);
+
+    assert_string_equal(data->output, "err Cannot execute backup get command, unable to open 'backup/db' folder");
+    assert_int_equal(result, OS_INVALID);
+
+    os_free(query);
+}
+
+void test_wdb_parse_global_get_backup_success(void **state) {
+    test_struct_t *data  = (test_struct_t *)*state;
+    int result = OS_INVALID;
+    char *query = NULL;
+    cJSON* j_backup = cJSON_Parse("[\"global.db-backup-TIMESTAMP\"]");
+
+    os_strdup("global backup get", query);
+    will_return(__wrap_wdb_open_global, data->wdb);
+    expect_string(__wrap__mdebug2, formatted_msg, "Global query: backup get");
+
+    will_return(__wrap_wdb_global_get_backups, j_backup);
+
+    result = wdb_parse(query, data->output, 0);
+
+    assert_string_equal(data->output, "ok [\"global.db-backup-TIMESTAMP\"]");
+    assert_int_equal(result, OS_SUCCESS);
+
+    os_free(query);
+}
+
+/* wdb_parse_global_restore_backup */
+
+void test_wdb_parse_global_restore_backup_invalid_syntax(void **state) {
+    test_struct_t *data  = (test_struct_t *)*state;
+    int result = OS_INVALID;
+    char *query = NULL;
+
+    os_strdup("global backup restore {INVALID_JSON}", query);
+    will_return(__wrap_wdb_open_global, data->wdb);
+    expect_string(__wrap__mdebug2, formatted_msg, "Global query: backup restore {INVALID_JSON}");
+
+    expect_string(__wrap__mdebug1, formatted_msg, "Invalid backup JSON syntax when restoring snapshot.");
+    expect_string(__wrap__mdebug2, formatted_msg, "JSON error near: NVALID_JSON}");
```

Comment:
I reviewed the code of the `wazuh-db` parser and the format string that causes this ugly message is used 262 times. I totally agree with you on the idea that this could be improved. But maybe we could make it in a separate issue and PR as it will cause failures in so many unit test cases.